File: TitleMenu_SC.lua
sum = AR:Summary({env = "OPN", bg = "Toast_BG"})
AreaTag("Medley_OPN")
tileMenuBG = AR:Movie({
  movie = "SEA_MainMenu_BG",
  source = Viewport.uiSize,
  onScreen = Viewport.uiSize,
  pauseOnLastFrame = true
})
if Flags.UI_SkipTitleFadeIn_FL then
  Flags.UI_SkipTitleFadeIn_FL = false
else
  coverBG = AR:Movie({
    movie = "SuperToast3_BG",
    source = Viewport.uiSize,
    onScreen = Viewport.uiSize,
    loop = true,
    z = 8
  })
  coverBG.animator[1].a = 1
  coverBG.animator[1].a:AddBlendLerp(1, 0)
end
local locked = false
titleOVL = AR:Overlay({
  ovl = "SEA_MainMenuTitle_OVL",
  source = Rect:New(0, 0, 808, 366),
  onScreen = Rect:New(108, 83, 916, 449),
  z = 4
})
nancyDrewOVL = AR:Overlay({
  ovl = "UI_MainMenuTitleND_OVL",
  source = Rect:New(0, 0, 808, 366),
  onScreen = Rect:New(108, 83, 916, 449),
  z = 3
})
mainMenuOVL = AR:Overlay({
  ovl = "UI_MainMenu_OVL",
  source = Rect:New(0, 0, 1024, 94),
  onScreen = Rect:New(0, 674, 1024, 768)
})
newGameButton = AR:Button({
  hs = AR:Hotspot({
    onScreen = Rect:New(43, 730, 169, 760),
    cursor = "MenuHot"
  }),
  overOvl = AR:Overlay({
    ovl = "UI_MainMenu_Rollover_OVL",
    source = Rect:New(2, 2, 128, 32),
    onScreen = Rect:New(43, 730, 169, 760)
  }),
  OnDown = function(this)
    Scene:Change(Scene.streamName, "Badges_SC")
  end
})
loadGameButton = AR:Button({
  hs = AR:Hotspot({
    onScreen = Rect:New(203, 730, 329, 760),
    cursor = "MenuHot"
  }),
  overOvl = AR:Overlay({
    ovl = "UI_MainMenu_Rollover_OVL",
    source = Rect:New(2, 34, 128, 64),
    onScreen = Rect:New(203, 730, 329, 760)
  }),
  OnDown = function(this)
    Scene:BeginStream({
      stream = "MainMenuPopup",
      scene = "UI_TM_Load_SC",
      captureInput = true
    })
  end
})
helpButton = AR:Button({
  hs = AR:Hotspot({
    onScreen = Rect:New(364, 730, 473, 760),
    cursor = "MenuHot"
  }),
  overOvl = AR:Overlay({
    ovl = "UI_MainMenu_Rollover_OVL",
    source = Rect:New(2, 66, 111, 96),
    onScreen = Rect:New(364, 730, 473, 760)
  }),
  OnDown = function(this)
    Scene:Change("TUT_Start_SC")
  end
})
optionsButton = AR:Button({
  hs = AR:Hotspot({
    onScreen = Rect:New(503, 730, 610, 760),
    cursor = "MenuHot"
  }),
  overOvl = AR:Overlay({
    ovl = "UI_MainMenu_Rollover_OVL",
    source = Rect:New(2, 98, 109, 128),
    onScreen = Rect:New(503, 730, 610, 760)
  }),
  OnDown = function(this)
    Scene:BeginStream({
      stream = "MainMenuPopup",
      scene = "UI_TM_Options_SC",
      captureInput = true
    })
  end
})
extrasButton = AR:Button({
  hs = AR:Hotspot({
    onScreen = Rect:New(635, 730, 742, 760),
    cursor = "MenuHot"
  }),
  overOvl = AR:Overlay({
    ovl = "UI_MainMenu_Rollover_OVL",
    source = Rect:New(2, 130, 109, 160),
    onScreen = Rect:New(635, 730, 742, 760)
  }),
  OnDown = function(this)
    Scene:BeginStream({
      stream = "MainMenuPopup",
      scene = "UI_Extras_SC",
      captureInput = true
    })
  end
})
moreNDButton = AR:Button({
  hs = AR:Hotspot({
    onScreen = Rect:New(761, 730, 884, 760),
    cursor = "MenuHot"
  }),
  overOvl = AR:Overlay({
    ovl = "UI_MainMenu_Rollover_OVL",
    source = Rect:New(2, 162, 125, 192),
    onScreen = Rect:New(761, 730, 884, 760)
  }),
  OnDown = function(this)
    Scene:BeginStream({
      stream = "MoreNDStream",
      scene = "UI_MoreND_SC",
      captureInput = true
    })
  end
})
quitButton = AR:Button({
  hs = AR:Hotspot({
    onScreen = Rect:New(890, 730, 1006, 760),
    cursor = "MenuHot"
  }),
  overOvl = AR:Overlay({
    ovl = "UI_MainMenu_Rollover_OVL",
    source = Rect:New(2, 194, 118, 224),
    onScreen = Rect:New(890, 730, 1006, 760)
  }),
  OnDown = function(this)
    Game:Exit()
  end
})

File: Ciftree/TitleMenu_SC.js
function TitleMenu_SC(anim = true) {
    let sum = AR.Summary({
        env: "OPN",
        bg: "toast_BG",
        z: 10
    });
    let tileMenuBG = AR.Movie({
        movie: "MID_MainMenu_BG",
        //source: Viewport.uiSize,
        onScreen: Viewport.uiSize,
        pauseOnLastFrame: true
    });
    let titleOVL = AR.Overlay({
        ovl: "MID_MainMenuTitle_OVL",
        source: [0, 0, 808, 366],
        onScreen: [108, 83, 916, 449],
        z: 4
    });
    let nancyDrewOVL = AR.Overlay({
        ovl: "UI_MainMenuTitleND_OVL",
        source: [0, 0, 808, 366],
        onScreen: [108, 83, 916, 449],
        z: 3
    });
    let mainMenuOVL = AR.Overlay({
        ovl: "UI_MainMenu_OVL",
        source: [0, 0, 1024, 94],
        onScreen: [0, 674, 1024, 768],
        z: 1
    });
    let newGameButton = AR.Button({
        hs: AR.Hotspot({
            onScreen: [43, 730, 169, 760],
            cursor: "MenuHot"
        }),
        overOvl: AR.Overlay({
            ovl: "UI_MainMenu_Rollover_OVL",
            source: [2, 2, 128, 32],
            onScreen: [43, 730, 169, 760]
        }),
        OnDown: function() {
            uiFrame.classList.remove("disabled");
            Scene.LetsGo("s2600");
            themeSound.pause();
            themeSound.currentTime = 0;
            //Scene.Change(Scene.streamName, "Badges_SC")
        }
    });
    let loadGameButton = AR.Button({
        hs: AR.Hotspot({
            onScreen: [203, 730, 329, 760],
            cursor:"MenuHot"
        }),
        overOvl: AR.Overlay({
            ovl: "UI_MainMenu_Rollover_OVL",
            source: [2, 34, 128, 64],
            onScreen: [203, 730, 329, 760]
        }),
        OnDown: function() {
            //Scene.BeginStream({
            //    stream: "MainMenuPopup",
            //    scene: "UI_TM_Load_SC",
            //    captureInput: true
            //})
        }
    });
    let helpButton = AR.Button({
        hs: AR.Hotspot({
            onScreen: [364, 730, 473, 760],
            cursor:"MenuHot"
        }),
        overOvl: AR.Overlay({
            ovl: "UI_MainMenu_Rollover_OVL",
            source: [2, 66, 111, 96],
            onScreen: [364, 730, 473, 760]
        }),
        OnDown: function() {
            Scene.Change("TUT_Start_SC")
        }
    })
    let optionsButton = AR.Button({
        hs: AR.Hotspot({
            onScreen: [503, 730, 610, 760],
            cursor:"MenuHot"
        }),
        overOvl: AR.Overlay({
            ovl: "UI_MainMenu_Rollover_OVL",
            source: [2, 98, 109, 128],
            onScreen: [503, 730, 610, 760]
        }),
        OnDown: function() {
            //Scene.BeginStream({
            //    stream: "MainMenuPopup",
            //    scene: "UI_TM_Options_SC",
            //    captureInput: true
            //})
        }
    });
    let extrasButton = AR.Button({
        hs: AR.Hotspot({
            onScreen: [635, 730, 742, 760],
            cursor:"MenuHot"
        }),
        overOvl: AR.Overlay({
            ovl: "UI_MainMenu_Rollover_OVL",
            source: [2, 130, 109, 160],
            onScreen: [635, 730, 742, 760]
        }),
        OnDown: function() {
            //Scene.BeginStream({
            //    stream: "MainMenuPopup",
            //    scene: "UI_Extras_SC",
            //    captureInput: true
            //})
        }
    });
    let moreNDButton = AR.Button({
        hs: AR.Hotspot({
            onScreen: [761, 730, 884, 760],
            cursor:"MenuHot"
        }),
        overOvl: AR.Overlay({
            ovl: "UI_MainMenu_Rollover_OVL",
            source: [2, 162, 125, 192],
            onScreen: [761, 730, 884, 760]
        }),
        OnDown: function() {
            //Scene.BeginStream({
            //    stream: "MoreNDStream",
            //    scene: "UI_MoreND_SC",
            //    captureInput: true
            //})
        }
    });
    let quitButton = AR.Button({
        hs: AR.Hotspot({
            onScreen: [890, 730, 1006, 760],
            cursor:"MenuHot"
        }),
        overOvl: AR.Overlay({
            ovl: "UI_MainMenu_Rollover_OVL",
            source: [2, 194, 118, 224],
            onScreen: [890, 730, 1006, 760]
        }),
        OnDown: function() {
            //Game.Exit()
        }
    });
    var themeSound = AR.Sound({
        sounds: "MainTheme_SFX",
        channel: "Theme",
        volume: 0.75,
        loop: true,
        active: !anim
    });
    let coverBG = AR.Movie({
        movie: "toast_BG",
        id: "coverBG",
        source: Viewport.uiSize,
        onScreen: Viewport.uiSize,
        loop: true,
        z: 8,
        duration: 2,
        opacity: 0,
        animationName: "fadeOut"
    });
    var twoSeconds = AR.Timer({
        duration: 2,
        OnEnd: function() {
            themeSound.play();
        }
    });
    let returnArray = [sum, tileMenuBG, titleOVL, nancyDrewOVL, mainMenuOVL, newGameButton, loadGameButton, helpButton, optionsButton, extrasButton, moreNDButton, quitButton];
    if(anim == true) {
        returnArray.push(coverBG);
        twoSeconds.start();
        
    }
    return returnArray;
}